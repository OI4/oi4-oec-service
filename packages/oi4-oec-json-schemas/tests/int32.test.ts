import { matchersWithOptions } from 'jest-json-schema'
import schema from '../src/schemas/dataTypes/int32.schema.json'

expect.extend(
  matchersWithOptions({
    verbose: true,
  })
)

describe('int32 schema', () => {
  it('validate schema', () => {
    expect(schema).toBeValidSchema()
  })

  it('test valid values', () => {
    expect(-2147483648).toMatchSchema(schema)
    expect(-2147483647).toMatchSchema(schema)
    expect(-1).toMatchSchema(schema)
    expect(0).toMatchSchema(schema)
    expect(1).toMatchSchema(schema)
    expect(1.0).toMatchSchema(schema)
    expect(2147483646).toMatchSchema(schema)
    expect(2147483647).toMatchSchema(schema)
  })

  it('test invalid values', () => {
    expect(-2147483649).not.toMatchSchema(schema)
    expect(-1.5).not.toMatchSchema(schema)
    expect(2147483648).not.toMatchSchema(schema)

    expect(() => {
      expect(-2147483649).toMatchSchema(schema)
    }).toThrowErrorMatchingSnapshot('-2147483649')
    expect(() => {
      expect(1.5).toMatchSchema(schema)
    }).toThrowErrorMatchingSnapshot('1.5')
    expect(() => {
      expect(2147483648).toMatchSchema(schema)
    }).toThrowErrorMatchingSnapshot('2147483648')
  })
})
