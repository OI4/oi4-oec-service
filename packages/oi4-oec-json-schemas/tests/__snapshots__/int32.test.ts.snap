// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`int32 schema test invalid values: -2147483649 1`] = `
"expect(received).toMatchSchema(schema)

received
  must be >= -2147483648
    Path:     dataTypes/int32.schema.json#/minimum
"
`;

exports[`int32 schema test invalid values: 1.5 1`] = `
"expect(received).toMatchSchema(schema)

received
  must be integer
    Path:     dataTypes/int32.schema.json#/type
"
`;

exports[`int32 schema test invalid values: 2147483648 1`] = `
"expect(received).toMatchSchema(schema)

received
  must be <= 2147483647
    Path:     dataTypes/int32.schema.json#/maximum
"
`;
